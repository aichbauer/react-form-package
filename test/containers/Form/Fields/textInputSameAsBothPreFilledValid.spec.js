import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with two text inputs and pre filled value | sameAs | both should be valid', () => {
  const myComponent = (
    <Form>
      <Field type="text" sameAs="text2" id="text" required value="Test" />
      <Field type="text" sameAs="text" id="text2" required value="Test" />
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  const state = wrapper.state();

  expect(state.data.text.value).toBe('Test');
  expect(state.data.text.valid).toBe(true);
  expect(state.data.text2.value).toBe('Test');
  expect(state.data.text2.valid).toBe(true);

  expect(tree).toMatchSnapshot();
});
