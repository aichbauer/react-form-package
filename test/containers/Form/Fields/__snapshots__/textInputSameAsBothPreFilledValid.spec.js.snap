// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with two text inputs and pre filled value | sameAs | both should be valid 1`] = `
<form
  enctype="application/x-www-form-urlencoded"
>
  <input
    className="rfp-input"
    id="text"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    sameAs="text2"
    type="text"
    valid={false}
    value="Test"
  />
  <input
    className="rfp-input"
    id="text2"
    invalid={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    required={true}
    sameAs="text"
    type="text"
    valid={false}
    value="Test"
  />
</form>
`;
